### PR TITLE
Follow team list

### PR DESCRIPTION
**Pull request by thomaspaulin**
Just doing a single team for now. When I have more of an understanding of Angular in general I'll look into implementing multiple teams and hopefully in a reactive form manner.

### CODE REVIEW

File: src/app/follow-team-list/follow-team-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  ngOnInit() {
+    // this.followForm.valueChanges
+    //   .debounceTime(300)
+    //   .subscribe(data => console.log(data));
+  }
+
+  toggleFollow(event: MdCheckboxChange) {
+    // todo make this more functional like when the form controls and groups etc are sorted
+    let team = TEAMS[parseInt(event.source.id) - 1];
```
Comment (oh-wo) [inline src/app/follow-team-list/follow-team-list.component.ts]: NIT: Prefer `const`

File: src/app/follow-team-list/follow-team-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    let index = this.following.indexOf(team);
+    if(index > -1) {
+      let spliced = this.following.splice(index);
+      console.log(spliced.length);
+      if(spliced.length > 1) {
+        // for testing debouncing worked
+        throw new Error("Too many items removed!!");
```
Comment (oh-wo) [inline src/app/follow-team-list/follow-team-list.component.ts]: - Unlike java, in javascript (typescript?!) try to stay away from throwing errors. 
- Maybe times are changing and this is more common now

File: src/app/follow-team-list/follow-team-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  following: Team[] = [];
+
+  constructor() {
+    // use a FormBuilder to take the reactive approach
+  }
+
+  ngOnInit() {
+    // this.followForm.valueChanges
+    //   .debounceTime(300)
```
Comment (oh-wo) [inline src/app/follow-team-list/follow-team-list.component.ts]: Why is debouncing needed? Lets talk about this - I want to understand.

File: src/app/follow-team-list/follow-team-list.component.ts (hunk 1 above)
Comment (oh-wo) [inline src/app/follow-team-list/follow-team-list.component.ts]: Could this be rewritten so `parseInt(event.source.id) - 1` isn't required?

File: src/app/app.component.html [hunk 4]
```diff
@@ -1,12 +1,13 @@
-<h1>{{title}}</h1>
+<!--<h1>{{title}}</h1>-->
 
-<hero-detail [hero]="selectedHero"></hero-detail>
+<!--<hero-detail [hero]="selectedHero"></hero-detail>-->
 
-<h2>My Heroes</h2>
-<ul class="heroes">
-  <li *ngFor="let hero of heroes"
-      [class.selected]="hero === selectedHero"
```
Comment (oh-wo) [inline src/app/app.component.html]: This is cool, I haven't seen it before. Does it bind the conditional statement to a class named `selected`?

File: src/app/follow-team-list/follow-team-list.component.ts (hunk 2 above)
Comment (thomaspaulin) [inline src/app/follow-team-list/follow-team-list.component.ts]: This was simply to make a case I didn't want to occur more visible while debugging.

File: src/app/app.component.html (hunk 4 above)
Comment (thomaspaulin) [inline src/app/app.component.html]: Yes. It adds the `selected` class when `hero === selectedHero` is true.

**Pull request merged**